doc/tutorial/search/index.md

```diff
@@ -0,0 +1,258 @@
+
+## Example
+
+Both FM- and Bi-FM-Indices internally use the implementation of the [Succinct Data Structure Library](https://github.com/simongog/sdsl-lite).<br>
+Conctructing a (Bi-)FM-Index is very simple as the template parameters are automatically deduced:
+
+\include doc/tutorial/search/basic_index.cpp
+
+You can also index text collections:
+
+\snippet doc/tutorial/search/small_snippets.cpp text_collection
+
+The indices can also be stored and loaded from disk.
+Both seqan3::fm_index::store and seqan3::fm_index::load return a `bool` indicating whether the operation was successful.
+
+\snippet doc/tutorial/search/small_snippets.cpp store
+
+\snippet doc/tutorial/search/small_snippets.cpp load
+
+Note that you have to templatise the `fm_index` for default construction. In contrast to the construction via a given `text`, the template cannot be deduced from the type of `text` by the compiler.
+
+\assignment{Assignment 1}
+You are given the text
+\code
+dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+\endcode
+Create an fm_index over the reference, store the index and load the index into a new fm_index object.
+Print whether the store and load operations were successful.
+\endassignment
+
+\solution
+\snippet doc/tutorial/search/solution1.cpp solution
+**Expected output:**
+~~~console
+Index stored successfully.
+Index loaded successfully.
+~~~
+\endsolution
+
+# Search
+
+Using an index, we can now conduct searches for a given query. In this part we will learn, how to search exactly, allow substitutions and indels, and how to configure what kind of results we want, e.g. all results vs. only the best result.
+
+## Terminology
+### Search scheme
+A search scheme is the manner of how a search is conducted. Depending on what kind of errors are allowed, there is an optimal way of search. For example, for 1 allowed substitution the trivial approach is to search for all possible permutations with one substitution. It is more efficient to divide the query into two blocks, search one block exactly and allow the error in the other block. Even though this has to be done twice (error in block 1 and error in block 2), this requires less lookups in the index than the trivial approach.
+### Optimal Search Scheme
+The scheme that requires the least amount of queries in the index is called an *optimal search scheme* and determining it is NP-hard.<br>
+SeqAn3 provides *optimal* search schemes for up to **4** errors for Bi-FM-Indices and provides sub-optimal, but still efficient search schemes otherwise.<br>
+The choice of the search scheme is done internally, but when using >= 2 errors it is advisable to construct a Bi-FM-Index since searches will be faster.<br>
```

Comment:
```suggestion

\note The choice of the search scheme is done internally you don't have to worry about it. You can only choose between the FM and Bi-FM-Index but when using >= 2 errors it is advisable to use a Bi-FM-Index since searches will be faster.<br>
```